[Bug]: Broke the ESM build by merging #929

### What happened?

You broke the ESM build by merging #929, ESM requires file extensions in the imports. I would revert #929 as from my understanding, You'd have to use a bundler to make use of tree-shaking, Discordx isn't bundled last I checked, It's built then published, Making the merge completely useless.

### Reproduction

Install latest version (11.7.2) and try and run it.

### Package

discordx

### Version

Stable

### Relevant log output

```Shell
> my-bot@1.0.0 dev /root/projects/teamwolfyta/my-bot
> ts-node-esm src/main.ts

/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:366
    throw new ERR_MODULE_NOT_FOUND(
          ^
CustomError: Cannot find module '/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/discordx@11.7.2_discord.js@14.8.0/node_modules/lodash/escapeRegExp' imported from /root/projects/teamwolfyta/my-bot/node_modules/.pnpm/discordx@11.7.2_discord.js@14.8.0/node_modules/discordx/build/esm/Client.js
    at finalizeResolution (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:366:11)
    at moduleResolve (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:801:10)
    at Object.defaultResolve (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/dist-raw/node-internal-modules-esm-resolve.js:912:11)
    at /root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/esm.ts:218:35
    at entrypointFallback (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/esm.ts:168:34)
    at /root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/esm.ts:217:14
    at addShortCircuitFlag (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/esm.ts:409:21)
    at resolve (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/esm.ts:197:12)
    at resolve (/root/projects/teamwolfyta/my-bot/node_modules/.pnpm/ts-node@10.9.1_cbfmry4sbbh4vatmdrsmatfg5a/node_modules/ts-node/src/child/child-loader.ts:15:39)
    at nextResolve (node:internal/modules/esm/loader:163:28)
 ELIFECYCLE  Command failed with exit code 1.
```


### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

--- a/packages/changelog/src/helper-tag.ts
+++ b/packages/changelog/src/helper-tag.ts
@@ -1,5 +1,5 @@
 import child from "child_process";
-import compact from "lodash/compact";
+import { compact } from "lodash";
 
 export function GetTags(match: string): string[] {
   const tags = compact(

--- a/packages/discordx/src/Client.ts
+++ b/packages/discordx/src/Client.ts
@@ -23,8 +23,7 @@ import {
   Client as ClientJS,
   InteractionType,
 } from "discord.js";
-import escapeRegExp from "lodash/escapeRegExp";
-import isArray from "lodash/isArray";
+import { escapeRegExp, isArray } from "lodash";
 
 import type {
   ApplicationCommandDataEx,

--- a/packages/discordx/src/logic/metadata/MetadataStorage.ts
+++ b/packages/discordx/src/logic/metadata/MetadataStorage.ts
@@ -5,7 +5,7 @@ import {
   ApplicationCommandOptionType,
   ApplicationCommandType,
 } from "discord.js";
-import findIndex from "lodash/findIndex";
+import { findIndex } from "lodash";
 
 import type { Method } from "../../decorators/classes/Method.js";
 import type {

--- a/packages/discordx/src/util/comparison.ts
+++ b/packages/discordx/src/util/comparison.ts
@@ -1,6 +1,5 @@
 import type { ApplicationCommand } from "discord.js";
-import isEqual from "lodash/isEqual";
-import omit from "lodash/omit";
+import { isEqual, omit } from "lodash";
 
 import type { DApplicationCommand } from "../decorators/index.js";
 import type { ApplicationCommandDataEx } from "../types/index.js";

--- a/packages/discordx/src/util/resolve-guilds.ts
+++ b/packages/discordx/src/util/resolve-guilds.ts
@@ -1,5 +1,4 @@
-import isEqual from "lodash/isEqual";
-import uniqWith from "lodash/uniqWith";
+import { isEqual, uniqWith } from "lodash";
 
 import type {
   Client,

--- a/packages/lava-queue/src/queue.ts
+++ b/packages/lava-queue/src/queue.ts
@@ -3,7 +3,7 @@ import type {
   Track,
   TrackResponse,
 } from "@discordx/lava-player";
-import shuffle from "lodash/shuffle";
+import { shuffle } from "lodash";
 
 import type { Player } from "./player.js";
 

--- a/packages/music/src/core/Queue.ts
+++ b/packages/music/src/core/Queue.ts
@@ -13,8 +13,7 @@ import {
   VoiceConnectionStatus,
 } from "@discordjs/voice";
 import type { Guild, StageChannel, VoiceChannel } from "discord.js";
-import pullAt from "lodash/pullAt";
-import shuffle from "lodash/shuffle";
+import { pullAt, shuffle } from "lodash";
 
 import type { CommonTrack, Player } from "../index.js";
 import { PlayerErrors, YoutubeTrack } from "../index.js";

--- a/packages/pagination/src/Pagination.ts
+++ b/packages/pagination/src/Pagination.ts
@@ -14,7 +14,7 @@ import {
   Message,
   MessageComponentInteraction,
 } from "discord.js";
-import cloneDeep from "lodash/cloneDeep";
+import { cloneDeep } from "lodash";
 
 import { GeneratePage } from "./functions/GeneratePage.js";
 import type { PaginationResolver } from "./Resolver.js";

